core/src/data/value/mod.rs

```diff
@@ -555,6 +555,218 @@ impl Value {
         }
     }
 
+    pub fn validate_shift_rhs(rhs: i64) -> Result<u32> {
+        match rhs.try_into() {
+            Ok(u) => Ok(u),
+            Err(_) => Err(ValueError::ImpossibleCast.into()),
+        }
+    }
+
+    pub fn shift_left(&self, rhs: &Value) -> Result<Value> {
+        use Value::*;
+
+        match (self, rhs) {
+            (I8(a), I64(b)) => {
+                let conversion_result = Self::validate_shift_rhs(*b);
```

Comment:
most of primitive integer types already have `checked_shl` so you don't need to convert the value into `u32`.
you can simply run `checked_shl` for `i8`, `i16`, ..